packages/leveldb/src/leveldb/internal.ts

```diff
@@ -36,20 +40,63 @@ export class LevelDBInternal<RxDocType> implements LevelDBStorageInternals<RxDoc
         if (LevelDBInternal.isLevelDBConstructor(this._options)) {
             this.db = this._options.level
         } else {
-            this.db = new Level(this._options.dbName, { valueEncoding: 'json' })
+            this.db = Level(this._options.dbPath, { valueEncoding: 'json' })
         }
     }
 
     async getDocuments(query: string[]): Promise<Map<string, RxDocumentData<RxDocType>>> {
         const docsInDbMap = new Map<string, RxDocumentData<RxDocType>>();
         if (query.length <= 0) {
             const db = await this.getInstance()
-            for await (const key of db.keys()) {
-                const value = await this.get(key);
-                if (value && !Array.isArray(value)) {
-                    docsInDbMap.set(key, value)
-                }
-            }
+              .catch(err => console.error('cant get db', err))
+
+            console.log('getDocuments')
+
+            /* ✓ this works, which suggests the db setup and running... */
+
+            // await db
+            //   .put('dog', { name: 'tony' })
+            //   .then(() => this.db.get('dog'))
+            //   .then(data => console.log('success', data))
+            //   .catch(console.error)
+
+
+            /* ✗ doesn't work */
+
+            // let isDone = false
+            // const map2 = new Map()
+            // const it = db.iterator()
+            //
+            // while (!isDone) {
+            //   await new Promise((resolve, reject) => {
+            //     it.next((err, key, value) => {
+            //       if (err) return reject(err)
+            //
+            //       if (!key) isDone = true
+            //       else map2.set(key, value)
+            //       resolve(null)
+            //     })
+            //   })
+            // }
+            //
+            // it.end(err => err && console.error(err))
+            // console.log(map2)
+
+
+            /* ✗ doesn't work */
+
+            await pull(
+              pullLevel.read(db),
+              pull.filter(row => !isArray(row.value)),
+              pull.map(row => {
+                docsInDbMap.set(row.key, row.value)
+                return
+              }),
+              pull.collectAsPromise()
+            )
+
+            console.log('getDocuments (done)', docsInDbMap)
+            // WIP - not reaching here?
```

Comment:
Still stuck here. Have confirmed the database can be put + get to at this point in the code.
Not seeing why it's getting stuck here?

I did some reading into Jest to try and understand if it's running things, swallowing errors etc.
I feel like it's fancy pants and might be obscuring problems

Reply:
yeah.. agree... and don't think tis worth investing time on finding out

Reply:
the test runner I use is the most vanilla you can get
- [`tape`](https://www.npmjs.com/package/tape)
- and use `tap-arc` as a printer

```bash
tape test/*.test.js | tap-arc
```

Reply:
mocha might be worth a look tho as the syntax is more like those jest tests with `describe`, `expect` etc.

One thing I don't know (and am keen to learn) is "what's the best way to run `.ts` tests". At some level it will always be "transpile then run" right?